ArduPlane/AP_Arming.cpp

```diff
@@ -136,6 +136,19 @@ bool AP_Arming_Plane::arm_checks(AP_Arming::Method method)
         gcs().send_text(MAV_SEVERITY_WARNING, "watchdog: Bypassing arming checks");
         return true;
     }
+
+    if (plane.g.fence_autoenable == 3) {
+        if (!plane.geofence_set_enabled(true, AUTO_TOGGLED)) {
+            gcs().send_text(MAV_SEVERITY_WARNING, "Fence: cannot enable for arming");
+            return false;
+        } else if (!plane.geofence_prearm_check()) {
+            plane.geofence_set_enabled(false, AUTO_TOGGLED);
+            return false;
+        } else {
+            gcs().send_text(MAV_SEVERITY_WARNING, "Fence: auto-enabled for arming");
+        }
+    }
+    
     // call parent class checks
     return AP_Arming::arm_checks(method);
```

Comment:
These can fail - in particular if your logging isn't working.

This leaves your vehicle in a changed state - your fence will still be enabled.